### PR TITLE
DT DQM: attempt to clean Legacy (Run1) DDU settings

### PR DESCRIPTION
**Pull request by jfernan2**
#### PR description:

This PR tries to clean DT DQM code from DDU settings belonging to Legacy (Run1). Also an empty plot in TM TriggerEfficiency is fixed by using an appropriate collection of digis

#### PR validation:

Tested with runTheMatrix.py -l limited -i all --ibeos

#### if this PR is a backport please specify the original PR:

No need to backport. Cleaning targeting Run3

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please abort

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1
Changes expected due to the cleaning

**Comment by fabiocos** [comment]
+1

**Comment by Martin-Grunewald** [comment]
Looks like this PR creates a problem in a Run-1 TSG test workflow:

```
----- Begin Fatal Exception 02-Oct-2019 07:58:11 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 256677 lumi: 380 event: 397024298 stream: 2
   [1] Calling method for module DTTriggerEfficiencyTask/'dtTriggerEfficiencyMo\
nitor'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: L1MuDTChambPhContainer
Looking for module label: twinMuxStage2Digis
Looking for productInstanceName: PhIn

----- End Fatal Exception -------------------------------------------------
```

**Comment by jfernan2** [comment]
@Martin-Grunewald  can you specify which wf? Is it a standard one? Run1 wf should not read that collection if Era treatment is applied, that's what prevents this line:
https://github.com/cms-sw/cmssw/pull/27996/files#diff-75afa721b61352b2086ab193486f0a68R28

> Looks like this PR creates a problem in a Run-1 TSG test workflow:
> 
> ```
> ----- Begin Fatal Exception 02-Oct-2019 07:58:11 CEST-----------------------
> An exception of category 'ProductNotFound' occurred while
>    [0] Processing  Event run: 256677 lumi: 380 event: 397024298 stream: 2
>    [1] Calling method for module DTTriggerEfficiencyTask/'dtTriggerEfficiencyMonitor'
> Exception Message:
> Principal::getByToken: Found zero products matching all criteria
> Looking for type: L1MuDTChambPhContainer
> Looking for module label: twinMuxStage2Digis
> Looking for productInstanceName: PhIn
> 
> ----- End Fatal Exception -------------------------------------------------
> ```

File: DQM/DTMonitorModule/python/dtTriggerEfficiencyTask_cfi.py [hunk 1]
```diff
@@ -30,6 +25,5 @@
 # Modify for running in run 2 2016 data
 #
 from Configuration.Eras.Modifier_run2_common_cff import run2_common
-run2_common.toModify( dtTriggerEfficiencyMonitor, checkRPCtriggers = cms.untracked.bool(False),processDDU = cms.untracked.bool(False))
-
+run2_common.toModify( dtTriggerEfficiencyMonitor, inputTagTM = cms.untracked.InputTag('twinMuxStage2Digis:PhIn'))
```
Comment (Martin-Grunewald) [inline DQM/DTMonitorModule/python/dtTriggerEfficiencyTask_cfi.py]: Should likely not be done for an L1T stage-1 workflow...

**Comment by Martin-Grunewald** [comment]
It is a Run-2 workflow but with L1T stage-1 setup (before L1T went to stage-2).

File: DQM/DTMonitorModule/python/dtTriggerEfficiencyTask_cfi.py (hunk 1 above)
Comment (jfernan2) [reply DQM/DTMonitorModule/python/dtTriggerEfficiencyTask_cfi.py]: Can you suggest an Era or equivalent modifier then? And a wf number to test it?
Thanks

**Comment by jfernan2** [comment]
> It is a Run-2 workflow but with L1T stage-1 setup (before L1T went to stage-2).

2016 onwards excluding 2015 perhaps?

**Comment by Martin-Grunewald** [comment]
Stage-1 L1T was used before stage-2 L1T, in 2015/16

File: DQM/DTMonitorModule/python/dtTriggerEfficiencyTask_cfi.py (hunk 1 above)
Comment (Martin-Grunewald) [reply DQM/DTMonitorModule/python/dtTriggerEfficiencyTask_cfi.py]: Hmm, perhaps `stage2L1Trigger` (so that it does not get applied to `stage1L1Trigger`)

**Comment by jfernan2** [comment]
Then I believe 
from Configuration.Eras.Modifier_run2_25ns_specific_cff import run2_25ns_specific
run2_25ns_specific.toModify(
is the right one. 
I will create a fix. Can you tell me a wf number to test it?
Apologies for the inconvenience

**Comment by Martin-Grunewald** [comment]
Many standard workflows (IB relvals) now fail with this, see:
https://cms-sw.github.io/relvalLogDetail.html#slc7_amd64_gcc820;CMSSW_11_0_X_2019-10-01-2300

**Comment by Martin-Grunewald** [comment]
Wouldn't   `stage2L1Trigger` be more targeted?

**Comment by jfernan2** [comment]
OK, I will try that then.

**Comment by jfernan2** [comment]
Fixing this problem in: PR #28105
Thanks @Martin-Grunewald for spotting

**Pull request merged**